File: part7_21_provider/src/main/java/com/example/part7_21_provider/DBHelper.java
package com.example.part7_21_provider;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class DBHelper extends SQLiteOpenHelper {
    public static final int DATABASE_VERSION=6;

    public DBHelper(Context context){
        super(context, "datadb", null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase sqLiteDatabase) {
        String tableSql="create table tb_data ("+
                "_id integer primary key autoincrement," +
                "name not null," +
                "phone)";

        sqLiteDatabase.execSQL(tableSql);

        sqLiteDatabase.execSQL("insert into tb_data (name,phone) values ('박찬호','0101111111')");
        sqLiteDatabase.execSQL("insert into tb_data (name,phone) values ('류현진','0102222222')");
        sqLiteDatabase.execSQL("insert into tb_data (name,phone) values ('오승환','0103333333')");
        sqLiteDatabase.execSQL("insert into tb_data (name,phone) values ('김현수','0105555555')");
        sqLiteDatabase.execSQL("insert into tb_data (name,phone) values ('추신수','0106666666')");
    }

    @Override
    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {
        if(i1 == DATABASE_VERSION){
            sqLiteDatabase.execSQL("drop table tb_data");
            onCreate(sqLiteDatabase);
        }
    }
}

File: part11_34/src/main/java/com/example/part11_34/DBHelper.kt
package com.example.part11_34

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class DBHelper (context : Context) : SQLiteOpenHelper(context, "datadb", null, 1) {
    override fun onCreate(p0: SQLiteDatabase?) {
        val tableSql = "create table tb_data ("+"_id integer primary key autoincrement," + "name not null," + "date)"

        p0?.execSQL(tableSql)
        p0?.execSQL("insert into tb_data (name,date) values ('elle smith','2019-07-01')")
        p0?.execSQL("insert into tb_data (name,date) values ('jessica scott','2019-07-01')")
        p0?.execSQL("insert into tb_data (name,date) values ('jack thomas','2019-07-01')")
        p0?.execSQL("insert into tb_data (name,date) values ('mike willis','2019-06-30')")
        p0?.execSQL("insert into tb_data (name,date) values ('eden john ','2019-06-28')")
        p0?.execSQL("insert into tb_data (name,date) values ('paul anderson','2019-06-28')")
        p0?.execSQL("insert into tb_data (name,date) values ('jenny miller','2019-06-28')")
    }

    override fun onUpgrade(p0: SQLiteDatabase?, p1: Int, p2: Int) {
        p0!!.execSQL("drop table tb_data")
        onCreate(p0)
    }
}